lib/route-enumeration.mjs

```
@@ -398,16 +400,26 @@ function discoverAppRoutes() {
-    const hasPage = entries.some(
+    const pageEntry = entries.find(
       (e) => e.isFile() && (e.name === 'page.tsx' || e.name === 'page.mdx' || e.name === 'page.jsx')
     )
-    if (hasPage) {
+    if (pageEntry) {
       const href = '/' + segments.join('/')
+      let sitemap = true
+      try {
+        const source = fs.readFileSync(path.join(dir, pageEntry.name), 'utf8')
+        sitemap =
+          !/\bnoindex\s*:\s*true\b/.test(source) &&
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)